### DISCUSSION TITLE
[Docker] [6.12.1] mympd ignores [mpd] host, port, and streamport

### DISCUSSION THREAD

**Discussion by helkaluin** [jcorporation/myMPD#459] [Q&A] [answered]
myMPD version: 6.12.1

**Describe the bug**

Logs indicate that both mpdworker and mpdclient are using a hard-coded 127.0.0.1:6600 instead of the host and port defined in mympd.conf. mympd.conf is definitely read since webport can be changed. On top of that, setting streamport returns "Unkown config option: mpd - streamport" in the logs.

**To Reproduce**

1. Use the official docker-compose
2. Edit mympd.conf
3. Run docker-compose up

**Server plattform (please complete the following information):**
 - OS: Ubuntu
 - OS version: 20.04
 - Plattform: x86_64
 - MPD version: 0.21.23

**Debug logs (please attach if it can be usefull)**

```
mympd    | NOTICE   mympd     Starting myMPD 6.12.1
mympd    | NOTICE   mympd     Libmympdclient 1.0.5 based on libmpdclient 2.20.0
mympd    | NOTICE   mympd     Mongoose 6.18
mympd    | NOTICE   mympd     Parsing config file: /etc/mympd.conf
mympd    | WARN     mympd     Unkown config option: mpd - streamport
mympd    | NOTICE   mympd     Setting loglevel to NOTICE
mympd    | NOTICE   mympd     Localstate dir: "/var/lib/mympd"
mympd    | NOTICE   mympd     Listening on http port 8075
mympd    | NOTICE   mympd     Droping privileges to mympd
mympd    | NOTICE   mympd     Smartpls dir: "/var/lib/mympd/smartpls"
mympd    | NOTICE   mympd     State dir: "/var/lib/mympd/state"
mympd    | NOTICE   mympd     Pics dir: "/var/lib/mympd/pics"
mympd    | NOTICE   mympd     Empty dir: "/var/lib/mympd/empty"
mympd    | NOTICE   mympd     Scripts dir: "/var/lib/mympd/scripts"
mympd    | NOTICE   mympd     Covercache dir: "/var/lib/mympd/covercache"
mympd    | NOTICE   mympd     MPD playlists dir: "/playlists"
mympd    | NOTICE   mympd     Starting mympd api thread
mympd    | NOTICE   mympdapi  Reading states
mympd    | NOTICE   mympd     Starting mpd worker thread
mympd    | NOTICE   mympd     Starting mpd client thread
mympd    | NOTICE   mympd     Starting webserver thread
mympd    | NOTICE   mpdworker Starting mpd_worker
mympd    | NOTICE   mpdworker MPD worker connecting to 127.0.0.1:6600
mympd    | NOTICE   mpdclient Starting mpd_client
mympd    | ERROR    mpdworker MPD worker connection: Connection refused
mympd    | NOTICE   mpdclient MPD connecting to 127.0.0.1:6600
mympd    | ERROR    mpdworker MPD worker connection failed
mympd    | ERROR    mpdclient MPD connection: Connection refused
mympd    | ERROR    mpdclient MPD connection failed
```

**Comment by jcorporation** [comment]
Please read the documentation, from https://github.com/jcorporation/myMPD/wiki/Configuration:

> myMPD uses /etc/mympd.conf as default configuration file. Path to this file is an optional argument to the mympd executable and can be omitted. Configuration will be overwritten by environment variables and by the settings in the web ui. Settings from the web ui are saved in individual files under /var/lib/mympd/state/. Deleting this files and restarting myMPD resets the web ui configuration.

I know this behavior is not obvious and it will be changed in the 7.1.0 release of myMPD.

If this does not work, please provide debug log files.

**Comment by helkaluin** [comment]
Many thanks. I did delete the /state files but then the defaults get propagated again (ignoring the .conf). Manually editing the /state files had the intended results.

**Comment by jcorporation** [comment, marked as answer]
The default Dockerfile sets mpd host and port to 127.0.0.1:6600 and overwrite the mympd.conf

streamport is a new setting for myMPD v7.0.0.
